tiledb/sm/misc/cancelable_tasks.h

```diff
@@ -0,0 +1,156 @@
+/**
+ * @file   cancelable_tasks.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2018 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the CancelableTasks class.
+ */
+
+#ifndef TILEDB_CANCELABLE_TASKS_H
+#define TILEDB_CANCELABLE_TASKS_H
+
+#include <functional>
+#include <iostream>
+
+#include "tiledb/sm/misc/status.h"
+#include "tiledb/sm/misc/thread_pool.h"
+
+namespace tiledb {
+namespace sm {
+
+class CancelableTasks {
```

Comment:
Also this isn't templated, so it should probably go in its own `.cc` file.

Reply:
Done -- I wasn't sure what our criteria was for using strictly header files only for small files. I'll stick with .cc files from now on, unless it will be a widely used template class.